src/shared/install.c

```diff
@@ -2625,7 +2628,10 @@ int unit_file_disable(
         if (r < 0)
                 return r;
 
-        config_path = (flags & UNIT_FILE_RUNTIME) ? paths.runtime_config : paths.persistent_config;
+        if (FLAGS_SET(flags, UNIT_FILE_PORTABLE))
+                config_path = FLAGS_SET(flags, UNIT_FILE_RUNTIME) ? paths.runtime_attached : paths.persistent_attached;
+        else
+                config_path = FLAGS_SET(flags, UNIT_FILE_RUNTIME) ? paths.runtime_config : paths.persistent_config;
```

Comment:
given this code is now used here multiple times and pretty complex, please add a function that takes `paths` and `flags` as param and returns the right path

Reply:
done